sys/include/net/gnrc/ipv6/ext.h

```diff
@@ -66,6 +66,33 @@ gnrc_pktsnip_t *gnrc_ipv6_ext_demux(gnrc_pktsnip_t *pkt, unsigned nh);
 gnrc_pktsnip_t *gnrc_ipv6_ext_build(gnrc_pktsnip_t *ipv6, gnrc_pktsnip_t *next,
                                     uint8_t nh, size_t size);
 
+#if     defined(MODULE_GNRC_IPV6_EXT) || defined(DOXYGEN)
+/**
+ * @brief   Processes a packet's extension headers.
+ *
+ * @param[in] pkt       An IPv6 packet in receive order.
+ * @param[in,out] nh    **In:** The @ref net_protnum of gnrc_pktsnip_t::data of
+ *                      @p pkt (i.e. the first extension header to be
+ *                      processed). <br />
+ *                      **Out:** The @ref net_protnum of header in
+ *                      gnrc_pktsnip_t::data of @p pkt. The extension headers
+ *                      are now marked, so their data can be found in
+ *                      gnrc_pktsnip_t::next of @p pkt and the following. <br />
+ *                      If the return value is NULL, the value of @p nh is
+ *                      undefined.
+ *
+ * @return  @p pkt with all extension headers marked until the first
+ *          non-extension header.
+ * @return  NULL, if packet was consumed by the extension header handling.
+ * @return  NULL, on error. @p pkt is released with EINVAL in that case.
+ */
+gnrc_pktsnip_t *gnrc_ipv6_ext_process_all(gnrc_pktsnip_t *pkt,
+                                          uint8_t *nh);
+#else   /* defined(MODULE_GNRC_IPV6_EXT) || defined(DOXYGEN) */
+/* NOPs to make the usage code more readable */
+#define gnrc_ipv6_ext_process_all(pkt, first_nh)    (pkt);
```

Comment:
This is a little bit misleading IMO. This macro looks like a function and someone could spend some time looking to the actual implementation of `gnrc_ipv6_ext_process_all` instead of the NOP.

Also I would prefer to move this macro to the C file and just leave only the prototype here.

I would do something like:

[gnrc_ipv6_ext.c]
```
#ifdef MODULE_GNRC_IPV6_EXT
#define GNRC_IPV6_EXT_PROCESS_ALL(pkt, first_nh) gnrc_ipv6_ext_process_all(pkt, nh)
#else 
#define GNRC_IPV6_EXT_PROCESS_ALL(pkt, first_nh) (pkt)
#endif
```

That way, it's clear GNRC_IPV6_EXT_PROCESS_ALL is a macro

Reply:
This is non-sensical. Then the user needs to use ifdefs anyway. The point of these NOP-macros is so you can use the functions in your code without worrying that the module is included or not.

Reply:
But your proposal uses NOPs as well. It just adds another level of indirection.

Reply:
To be more accurate, I meant I'm not convinced with mixing a macro and a function with the same name. I would prefer just use a second macro (as I described) or a weak-like symbol mechanism.

Or at least add a tiny note before the usage of this function indicating it resolves to `pkt` if `MODULE_GNRC_IPV6_EXT` is not included :)

Reply:
This mechanism is already used through-out the code and the behavior is the same as if `gnrc_ipv6_ext` would be in (but since the stack doesn't know about extension headers it is not handling them).

Reply:
ok, let's get it in

Reply:
Some examples:

https://github.com/RIOT-OS/RIOT/blob/b709e63581a85a504b1020a025ef6f0e73f80b1d/sys/include/net/gnrc/icmpv6/error.h#L95-L102

https://github.com/RIOT-OS/RIOT/blob/b709e63581a85a504b1020a025ef6f0e73f80b1d/sys/include/net/gnrc/netif/internal.h#L470-L471

The ironic thing: Some reviewers specifically asked me to do it this way...

Reply:
yes, I see. There are several ways to handle stubs. I'm fine ;)